Dockerfile

```diff
@@ -1,28 +1,33 @@
 FROM quay.io/centos/centos:stream8
 
 RUN dnf -y module install python39 && dnf -y install python39 python39-pip
-RUN mkdir /app
-ADD https://raw.githubusercontent.com/arcalot/arcaflow-plugin-template-python/main/LICENSE /app
-ADD example_plugin.py /app
-ADD test_example_plugin.py /app
-ADD requirements.txt /app
+RUN mkdir /app \
+ && chmod 700 /app
+ADD https://raw.githubusercontent.com/arcalot/arcaflow-plugin-template-python/main/LICENSE /app/
+COPY README.md /app/
+COPY poetry.lock /app/
+COPY pyproject.toml /app/
+COPY example_plugin.py /app/
+COPY test_example_plugin.py /app/
 WORKDIR /app
 
-RUN pip3 install -r requirements.txt
+RUN python3.9 -m pip install poetry \
+ && python3.9 -m poetry config virtualenvs.create false \
+ && python3.9 -m poetry install --without dev
 
-RUN mkdir /htmlcov
-RUN pip3 install coverage
-RUN python3 -m coverage run test_example_plugin.py
-RUN python3 -m coverage html -d /htmlcov --omit=/usr/local/*
+RUN mkdir /htmlcov \
+ && python3.9 -m pip install coverage \
+ && python3 -m coverage run test_example_plugin.py \
```

Comment:
This appears to be using different versions of Python to install it, and run the tests.

Reply:
dang good catch! Yes you are right I will update